### PR TITLE
IntMap.mergeA: Fix effects sequencing with negative keys

### PR DESCRIPTION
**Pull request by sjakobi**
See #692.

### CODE REVIEW

**Comment by sjakobi** [comment]
@wrengr Could you review?

**Comment by sjakobi** [comment]
Why would `intmap-lazy-properties` pass, but `intmap-strict-properties` fail?! :open_mouth:

**Comment by sjakobi** [comment]
> Why would `intmap-lazy-properties` pass, but `intmap-strict-properties` fail?!

Ah, the strict version of `traverseWithKey` still needs the fix from https://github.com/haskell/containers/pull/688. I'll wait until that has been merged then.

File: containers-tests/tests/intmap-properties.hs [hunk 1]
```diff
@@ -935,6 +938,19 @@ prop_mergeWithKeyModel xs ys
           -- warnings are issued if testMergeWithKey gets inlined.
           {-# NOINLINE testMergeWithKey #-}
 
+-- This uses the instance
+--     Monoid a => Applicative ((,) a)
+-- to test that effects are sequenced in ascending key order.
+prop_mergeA_effects :: [Int] -> [Int] -> Property
```
Comment (sjakobi) [inline containers-tests/tests/intmap-properties.hs]:
```suggestion
prop_mergeA_effects :: UMap -> UMap -> Property
```

**Comment by treeowl** [comment]
I've merged #688, so hopefully you can proceed here.

**Comment by treeowl** [comment]
*Sigh*. I tried to rebase with the web client but clearly that didn't work. You'll have to do it by hand.

**Comment by sjakobi** [comment]
> _Sigh_. I tried to rebase with the web client but clearly that didn't work. You'll have to do it by hand.

It looks fine to me. You can just squash the commits once you merge it.

**Comment by sjakobi** [comment]
@treeowl Is this fine to merge?

It would really appreciate a review from @wrengr but I'm not sure whether we ought to wait for that.

I think it would be great to get this (and #688) into GHC 8.10.1. How fast do we need to act for that @bgamari?

**Comment by treeowl** [comment]
Please do me a favor and squash/rebase so I can just merge from the web interface.

**Comment by treeowl** [comment]
While it would be nice to hear from Wren, she hasn't been available much for quite a while; we'll have to rely on our tests and hope for the best. I'll merge this as soon as CI completes.

**Comment by sjakobi** [comment]
> Please do me a favor and squash/rebase so I can just merge from the web interface.

Done!

**Comment by treeowl** [comment]
Thanks a lot!

**Pull request merged**